src/combinator.rs

```diff
@@ -57,6 +57,65 @@ impl <P: Parser> Iterator for Iter<P> {
 }
 
 #[derive(Clone)]
+pub struct ChoiceSlice<'a, P>(&'a [P])
+    where P: Parser + 'a;
+
+impl <'a, I, O, P> Parser for ChoiceSlice<'a, P>
+    where I: Stream
+        , P: Parser<Input=I, Output=O> + Clone +  'a {
+    type Input = I;
+    type Output = O;
+    fn parse_state(&mut self, input: State<I>) -> ParseResult<O, I> {
+        if self.0.len() == 0 { // err: slice empty of parsers
+            return Err(Consumed::Empty(ParseError { position: input.position.clone(), errors: vec![] }))
```

Comment:
Use the ParseError::new function instead. Since the fields are all public it is possible to construct like this but I defined the new function in some parts to make sure that there is always an error (no empty errors vector).

In this case something like `Error::Message("parser choice used with no parsers")`